lib/build.py

```diff
@@ -199,12 +208,16 @@ def build_arg_get(arg):
    # Print summary & we’re done.
    if (ml.instruction_total_ct == 0):
       ch.FATAL("no instructions found: %s" % cli.file)
-   assert (ml.inst_prev.image_i + 1 == image_ct)  # should’ve errored already
+   if (cli.target is None):
+      assert ml.inst_prev.image_i + 1 == image_ct # should’ve errored already
+   elif (last_image is None):
+      ch.WARNING("--target: %s: not found; ignored" % cli.target)
    if (cli.force and ml.miss_ct != 0):
       ch.INFO("--force=%s: modified %d RUN instructions"
               % (cli.force, forcer.run_modified_ct))
    ch.INFO("grown in %d instructions: %s"
-           % (ml.instruction_total_ct, ml.inst_prev.image))
+           % (ml.instruction_total_ct - 1 if stop_build else ml.instruction_total_ct,
```

Comment:
I think you can probably make it so build always continues until `last_image` and `stop_build`. If `--target` given, set them accordingly; if not, then they are just the last stage. That would eliminate special cases like this.